pkg/datapath/loader/tc.go

```diff
@@ -25,9 +25,17 @@ func attachSKBProgram(device netlink.Link, prog *ebpf.Program, progName, bpffsDi
 	}
 
 	if tcxEnabled {
-		// Attach using tcx if available. This is seamless on interfaces with
-		// existing tc programs since attaching tcx disables legacy tc evaluation.
-		err := upsertTCXProgram(device, prog, progName, bpffsDir, parent)
+		var err error
+		if device.Type() == "netkit" {
```

Comment:
I think we can return early inside this if block. If it's a netkit device all the following code to allow switching between tcx and legacy tc doesn't really apply IMO because at this point there is no up/downgrade path if between veth and netkit devices.
```suggestion
		if device.Type() == "netkit" {
		  	// If the device is a netkit device, we know that netkit links are
			// supported, therefore use netkit instead of tcx. For all others like
			// host devices, rely on tcx.
			return upsertNetkitProgram(device, prog, progName, bpffsDir, parent)
		}
		err := upsertTCXProgram(device, prog, progName, bpffsDir, parent)
		...
```